tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -1454,38 +1454,169 @@ Status Converter::RenameAndMarkOutputTensors(
+  TftrtAlgorithmSelector() {
+    int64_t trt_algorithm_id;
+    int64_t null_tactic = std::numeric_limits<int64_t>::min();
+    TF_CHECK_OK(tensorflow::ReadInt64FromEnvVar("TF_TRT_FIXED_ALGORITHM_ID",
+                                                /*default_val=*/null_tactic,
+                                                &trt_algorithm_id));
+    // Populate allowed tactics. Used to force select tactics for any layer.
+    // TODO: accepting a list from the environment variable
+    if (trt_algorithm_id != null_tactic) {
+      allowed_tactics.insert(trt_algorithm_id);
+    }
+
+#if !IS_TRT_VERSION_GE(8, 0, 0, 0)
+    disallowed_tactics = {
+      // turing_fp16_s1688cudnn_fp16_128x128_ldg8_relu_f2f_exp_medium_nhwc_gelu_tn_v1
+      -5927686925093575778,
+      //turing_fp16_s1688cudnn_fp16_128x128_ldg8_relu_f2f_exp_interior_nhwc_gelu_tn_v1
+      -3848538574386518527,
+      //turing_fp16_s1688cudnn_fp16_128x128_ldg8_relu_f2f_exp_small_nhwc_gelu_tn_v1
+      -959009792490796596
+    };
+#endif  // !IS_TRT_VERSION_GE(8, 0, 0, 0)
+
+    // Check whether allowed and disallowed tactics are disjointed.
+    if (allowed_tactics.size() > 0 && disallowed_tactics.size() > 0) {
+      std::vector<int64_t> intersection;
+      for (int64_t allowed : allowed_tactics) {
+        if (disallowed_tactics.count(allowed) > 0) {
+          intersection.emplace_back(allowed);
+        }
+      }
+      if (intersection.size() > 0) {
+        VLOG(1) << "Intersection of allowed and disallowed tactics found "
+                   "non-empty.";
+
+        VLOG(1) << "List of allowed tactics:";
+        VLOG(1) << absl::StrJoin(allowed_tactics, ", ");
 
+        VLOG(1) << "List of disallowed tactics:";
+        VLOG(1) << absl::StrJoin(disallowed_tactics, ", ");
+
+        VLOG(1) << "List of tactics common to both allowed and disallowed "
+                   "tactics:";
+        VLOG(1) << absl::StrJoin(intersection, ", ");
+
+        VLOG(1) << "Aborting.";
+        assert(0);
+      }
+    }
+  }
   // Returns value in [0, nbChoices] for a valid algorithm.
   int32_t selectAlgorithms(const nvinfer1::IAlgorithmContext& algoContext,
                            const nvinfer1::IAlgorithm* const* algoChoices,
-                           int32_t nbChoices,
-                           int32_t* selection) noexcept override {
+                           int32_t nbChoices, int32_t* selection) override {
+
+    std::unordered_set<int64_t> cherrypicked_tactics;
+    std::unordered_set<int64_t> rejected_tactics;
+
     // TensorRT always provides more than zero number of algorithms
     // in selectAlgorithms.
     assert(nbChoices > 0);
 
-    // making sure that the requested TRT algorithm ID doesn't go above the
-    // max value accepted.
-    selection[0] = std::min(algorithm_id_, nbChoices);
-    return 1;
+    // Disable all disallowed tactics
+    for (auto i = 0; i < nbChoices; i++) {
+      if (algoChoices[i] == nullptr) {
```

Comment:
initialize optional<unit32_t> forced_algorithm_id as follows:
   if allowed_algorithm_ids (renamed from allowed_tactics) is not empty, assert it has one value, get the min 
   between is value and nbChoices for forced_algorithm_id

then inside the loop
   if forced_algorithm_id.has_value() and  forced_algorithm_id.value() != i {
         emit log
         continue to the next iteration of the loog
  }

Reply:
I tried to address your comment. Please let me know if it works now

Reply:
add:

if (forced_algorithm_id.has_value() && forced_algorithm_id.value() != i) {
   continue;
}